api/storage_service.cc

```diff
@@ -706,17 +706,18 @@ void set_storage_service(http_context& ctx, routes& r, sharded<service::storage_
         });
     });
 
-    ss::force_keyspace_cleanup.set(r, [&ctx, &ss](std::unique_ptr<request> req) {
-        auto keyspace = validate_keyspace(ctx, req->param);
-        auto column_families = split_cf(req->get_query_param("cf"));
+    register_path_routine(ctx, r, "keyspace_cleanup", ss::force_keyspace_cleanup, {}, [&ctx, &ss] (std::unordered_map<sstring, sstring>&& val) {
+        sstring keyspace = val["ks"];;
+        sstring cf = val["cf"];
+        auto column_families = split_cf(cf);
         if (column_families.empty()) {
             column_families = map_keys(ctx.db.local().find_keyspace(keyspace).metadata().get()->cf_meta_data());
         }
         return ss.local().is_cleanup_allowed(keyspace).then([&ctx, keyspace,
                 column_families = std::move(column_families)] (bool is_cleanup_allowed) mutable {
             if (!is_cleanup_allowed) {
-                return make_exception_future<json::json_return_type>(
-                        std::runtime_error("Can not perform cleanup operation when topology changes"));
+                throw std::runtime_error("Can not perform cleanup operation when topology changes");
+                return make_ready_future<>();
```

Comment:
unreachable code; if the return type needs to be specified then it would look nicer in the lambda's "heading"

Reply:
Why is this not reachable?

Reply:
The `return` statement, right? Or I'm blind? (that's also possible)

Reply:
you're absolutely right, it turns out I'm blind